src/components/Lista/ItemListFiltr.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState } from 'react'
+import { HiArrowSmDown, HiArrowSmUp } from 'react-icons/hi'
+import { motion, AnimatePresence } from "framer-motion"
+
+const ItemListFiltr = ({ setFullList, fullList }) => {
+
+
+    const loadListFromServer = () => {
+        fetch('http://localhost:3001/actualList')
+            .then(response => response.json())
+            .then(data => {
+                // Przetwórz dane tak, aby pasowały do formatu twojej aplikacji
+                const newList = data[0].list.map(item => ({
+                    name: item.name,
+                    quantity: item.quantity,
+                    price: item.price,
+                    status: item.status
+                }));
+
+                // Zaktualizuj stan fullList, aby odzwierciedlić pobrane dane
+                setFullList(newList);
+            })
+            .catch(error => console.log('error', error));
+    };
+
+
+    const [filter, setFilter] = useState(false)
+    const toggleFilter = () => {
+        setFilter(!filter)
+    }
+
+    const sortFullListByLowestPrice = () => {
+        const newList = [...fullList].sort((a, b) => parseFloat(a.price) - parseFloat(b.price))
+        setFullList(newList)
+    }
+
+    const sortFullListByHighestPrice = () => {
+        const newList = [...fullList].sort((a, b) => parseFloat(b.price) - parseFloat(a.price))
+        setFullList(newList)
+    }
+
+    const sortFullListByHighestQuantity = () => {
+        const newList = [...fullList].sort((a, b) => parseFloat(b.quantity) - parseFloat(a.quantity))
+        setFullList(newList)
+    }
+
+    const sortFullListByLowestQuantity = () => {
+        const newList = [...fullList].sort((a, b) => parseFloat(a.quantity) - parseFloat(b.quantity))
+        setFullList(newList)
+    }
```

Comment:
tutaj nie jestem pewien ale wydaje się że dałoby się to zrefaktorować aby kod był bardziej DRY